pkg/utils/o11y/controller.go

```diff
@@ -83,3 +88,68 @@ func (h *SuiteController) GetRegexPodNameWithSize(podNameRegex string, results [
 
 	return podNameWithSize, nil
 }
+
+func (h *SuiteController) QuayImagePushPipelineRun(quayOrg, secret, namespace string) (*v1beta1.PipelineRun, error) {
+	pipelineRun := &v1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "pipeline-egress-",
+			Namespace:    namespace,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineSpec: &v1beta1.PipelineSpec{
+				Tasks: []v1beta1.PipelineTask{
+					{
+						Name: "buildah-quay",
+						TaskSpec: &v1beta1.EmbeddedTask{
+							TaskSpec: v1beta1.TaskSpec{
+								Steps: []v1beta1.Step{
+									{
+										Name:  "pull-and-push-image",
+										Image: "quay.io/redhat-appstudio/buildah:v1.28",
+										Env: []corev1.EnvVar{
+											{Name: "BUILDAH_FORMAT", Value: "oci"},
+											{Name: "STORAGE_DRIVER", Value: "vfs"},
+										},
+										Script: fmt.Sprintf(`#!/bin/sh
+												// srcImageRef="quay.io/redhat-appstudio/hacbs-test:latest"
+												authFilePath="/tekton/creds-secrets/%s/.dockerconfigjson"
+												destImageRef="quay.io/%s/o11y-workloads"
+												# Set Permissions
+												sed -i 's/^\s*short-name-mode\s*=\s*.*/short-name-mode = "disabled"/' /etc/containers/registries.conf
+												# Setting new namespace to run buildah - 2^32-2
+												echo 'root:1:4294967294' | tee -a /etc/subuid >> /etc/subgid
+												# Pull Image
+												echo -e "FROM quay.io/libpod/alpine:latest\nRUN dd if=/dev/urandom of=/100mbfile bs=1M count=100" > Dockerfile
+												unshare -Ufp --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah bud --tls-verify=false --no-cache -f ./Dockerfile -t "$destImageRef" .
+												IMAGE_SHA_DIGEST=$(buildah images --digests | grep ${destImageRef} | awk '{print $4}')
+												TAGGED_IMAGE_NAME="${destImageRef}:${IMAGE_SHA_DIGEST}"
+												buildah tag ${destImageRef} ${TAGGED_IMAGE_NAME}
+												buildah images
+												buildah push --authfile "$authFilePath" --disable-compression --tls-verify=false ${TAGGED_IMAGE_NAME}
+												echo "Successfully pushed Image"
+												# Sleep of 1m is for scraping interval
```

Comment:
please explain this - does this pipelinerun need to be running for the data to be successfully scraped?

Reply:
NO. it needs 30s for scraping interval to fetch the pipelinerun egress metrics after pipelinerun completion to get accurate result. so introduced 1m interval for worst case scenario for successful metrics fetch result.

Reply:
> ...to fetch the pipelinerun egress metrics after pipelinerun completion...

ok then it seems that there's no reason to have this hardcoded pause and keep this pipelinerun running then - I'd introduce a wait loop further in the test in which we could attempt to fetch the result, until it's successful, which could speed up the test

Reply:
I tried that as well. it fetched a lower value compared to the actual result of what is expected. so introduced this here. it seems consistent.

Reply:
I know it is not a good practice to have this in code. can I remove this from here for time being and introduce a wait loop in get metrics function when the function is rewritten? wdyt?

Reply:
Done

Reply:
To have accurate scraping, the pod that generates the traffic needs to stay alive between the time the traffic finished being generated and the time the scraping happens. @srivickynesh do we get that with a wait loop?

Reply:
well, the pods and pipeline runs get deleted only after the all tests are performed. the pod will be in a completed state when we fetch the results here.

Reply:
I'm not sure this is enough for the metrics to be collected.

@avi-biton do you recall if we had missing scraped data when pods didn't stay alive long enough to allow scraping, but were not deleted for a while after they died? I seem to recall that what affected full scraping was the time the pod remained actually alive and running.

Reply:
Yeah. It was about the init containers. @psturc We might need to have this sleep to get the correct data from the pod within the pipeline script.

Reply:
There is inaccurate information for short-lived pods observed when I tested it. even after the pod goes into a completed state. the pod should be alive & running until the scraping period.

Reply:
[FAILED] pipeline-egress-pvshz-buildah-quay-pod: 0.10 MB is not within the expected range. For this I introduced wait in metrics cmd. but as per yftach comment we will need sleep within the pipelinerun @psturc